### PR TITLE
fix Issue 14192 - Access Violation when assigning to shared AA

### PR DESCRIPTION
**Pull request by MartinNowak**
- the runtime expects an unqualified TypeInfo for the AA and crashes
  if it is wrapped inside a TypeInfo_Shared

[Issue 14192 – Access Violation when assigning to shared AA](https://issues.dlang.org/show_bug.cgi?id=14192)

### CODE REVIEW

File: src/e2ir.c [hunk 1]
```diff
@@ -4658,7 +4658,7 @@ elem *toElem(Expression *e, IRState *irs)
                 {
                     n1 = el_una(OPaddr, TYnptr, n1);
                     s = aaGetSymbol(taa, "GetY", 1);
-                    ti = toElem(getTypeInfo(taa->mutableOf(), NULL), irs);
+                    ti = toElem(getTypeInfo(taa->unSharedOf()->mutableOf(), NULL), irs);
```
Comment (rainers) [inline src/e2ir.c]: doesn't this combine to taa->nullAttributes()?

Comment (rainers) [inline src/e2ir.c]: Aah, just noticed that it would always create a new type, even if it is mutable and non-shared.

**Comment by rainers** [comment]
LGTM

**Comment by rainers** [comment]
Auto-merge toggled on

File: src/e2ir.c (hunk 1 above)
Comment (MartinNowak) [inline src/e2ir.c]: I just grepped for it and this is the pattern I found.

**Pull request merged**